lib/src/chunked_stream_reader.dart

```diff
@@ -129,22 +139,24 @@ class ChunkedStreamReader<T> {
     c.onListen = () => c.addStream(substream()).whenComplete(c.close);
     c.onCancel = () async {
       while (size > 0) {
```

Comment:
I assume that the `substream()` stream subscription has been cancelled at this point. Do check, carefully, that we actually wait for that to complete before calling `onCancel`, so you don't have two pieces of code reading he same input at the same time. (Just to be safe!)

Reply:
I think so:

https://github.com/dart-lang/sdk/blob/bf0086dbe205bc541734c5d16c3723b953df291f/sdk/lib/async/stream_controller.dart#L658-L659